common/utils.c

```diff
@@ -428,13 +428,24 @@ TDNFYesOrNo(
 
     if(!pArgs->nAssumeYes && !pArgs->nAssumeNo)
     {
-        pr_crit("%s", pszQuestion);
-        while ((opt = getchar()) == '\n' || opt == '\r');
-        opt = tolower(opt);
-        if (opt != 'y' && opt != 'n')
-        {
-            dwError = ERROR_TDNF_INVALID_INPUT;
-            BAIL_ON_TDNF_ERROR(dwError);
+        while(1) {
+            pr_crit("%s", pszQuestion);
+            char buf[256];
+            char *ret;
+
```

Comment:
do __fpurge(stdin) here

Reply:
I am not sure if that is needed. I experimented with it, but it does not seem to have an effect. I tried:
```
echo y | ./bin/tdnf install lsof
```
(which should always work).

Typing `y` and <return> while downloading cache data (to give me time), but it still accepts it. Which may not actually be a bad thing anyway.

It's a non-standard function, so it may not be a good idea.

Reply:
Yes, it's gcc specific. But I don't see a problem with using it.

see this once: https://stackoverflow.com/questions/20633284/strange-behaviour-of-c-codefgets-function

Reply:
I think the issue in that discussion is the `scanf()` not eating the line feed. We don't have that in tdnf.